File: SMeatSocialNetwork/SMeat.MODELS/Entities/Board.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SMeat.MODELS.Entities
{
    public class Board : Entity
    {
        public Board()
        {
            Id = Guid.NewGuid().ToString();
        }

        [Key]
        public override string Id { get; set; }

        public string Name { get; set; }

        public string Text { get; set; }

        public User MadeBy { get; set; }

        public DateTimeOffset MadeTime { get; set; } = DateTimeOffset.Now;

        public int RepliesCount { get; set; }

        public List<Reply> Replies { get; set; }

        public List<BoardLike> Likes { get; set; } = new List<BoardLike>();

        public List<BoardDislike> Dislikes { get; set; } = new List<BoardDislike>();
    }
}

File: SMeatSocialNetwork/SMeat.WEB/src/app/_models/board.ts
export class Board {

    id: string;
    name: string;
    text: string;
    likes;
    dislikes;

    constructor(id?: string, name?: string, text?: string, likes?, dislikes?) {
        this.id = id;
        this.name = name;
        this.text = text;
        this.likes = likes;
        this.dislikes = dislikes;
    }

}



